src/Microsoft.VisualStudio.Threading/AsyncLazy`1.cs

```diff
@@ -20,6 +25,11 @@ public class AsyncLazy<T>
     /// </summary>
     private static readonly object RecursiveCheckSentinel = new object();
 
+    /// <summary>
+    /// A value set on the <see cref="value"/> field when this object is disposed.
+    /// </summary>
+    private static readonly Task<T> DisposedSentinel = Task.FromException<T>(new ObjectDisposedException(nameof(AsyncLazy<T>)));
```

Comment:
Not sure if `nameof(AsyncLazy<T>)` is now the right parameter for the `ObjectDisposedException`, since that's not what's disposed.  However, if you tried to smuggle something like `nameof(T)` into the exception, you'd have a different exception for each `T`.